app/controllers/api/v1/Web.java

```diff
@@ -1451,11 +1464,324 @@ public static Result search() {
+      limit = Integer.valueOf(form.get("limit"));
+    }
+
+    if (offset < 0) {
+      offset = 0;
+    }
+
+    if (limit > SEARCH_APPLICATION_MAX_OFFSET) {
+      limit = SEARCH_APPLICATION_MAX_OFFSET;
+    } else if (limit <= 0) {
+      return ok(new Gson().toJson(new JsonObject()));
+    }
+
+    String sortBy = "severity";
+    boolean increasing = true;
+
+    String usernameString = form.get("usernames");
+    if(usernameString == null || usernameString.isEmpty()) {
+      JsonObject parent = new JsonObject();
+      parent.add(JsonKeys.USERRESULTS, new JsonObject());
+      return notFound(new Gson().toJson(parent));
+    }
+
+    List<String> usernames = Arrays.asList(usernameString.split(","));
+
+    Map<String,String> filterParamsForUserSummary = getFilterParamsForUserSummary();
+
+    if(form.get("sortKey")!=null) {
+      sortBy = form.get("sortKey");
+    }
+
+    if(form.get("increasing")!=null) {
+      increasing = Boolean.valueOf(form.get("increasing"));
+    }
+
+    JsonObject userResult = new JsonObject();
+    List<String> usernameQueryList = new ArrayList<String>();
+    for(int i=0;i<usernames.size();i++) {
+      usernameQueryList.add("username=:user" + i);
+    }
+
+    String usernameQueryString = String.join(" or ", usernameQueryList);
+
+    // by default, fetch data from last week
+    String finishedTimeBegin = String.valueOf(System.currentTimeMillis() - DAY*31);
+    String finishedTimeEnd = String.valueOf(System.currentTimeMillis());
+
+    if (Utils.isSet(filterParamsForUserSummary.get(Application.FINISHED_TIME_BEGIN))) {
+      finishedTimeBegin = filterParamsForUserSummary.get(Application.FINISHED_TIME_BEGIN);
+    }
+
+    if (Utils.isSet(filterParamsForUserSummary.get(Application.FINISHED_TIME_END))) {
+      finishedTimeEnd = filterParamsForUserSummary.get(Application.FINISHED_TIME_END);
+    }
+
+
+    StringBuilder timeFilterStringBuilder = new StringBuilder();
+    if(finishedTimeBegin!=null) {
+      timeFilterStringBuilder.append("finish_time");
+      timeFilterStringBuilder.append(">=");
+      timeFilterStringBuilder.append(parseTime(String.valueOf(finishedTimeBegin)));
+      if(finishedTimeEnd!=null) {
+        timeFilterStringBuilder.append(" and ");
+      }
+    }
+
+    if(finishedTimeEnd!=null) {
+      timeFilterStringBuilder.append("finish_time");
+      timeFilterStringBuilder.append("<=");
+      timeFilterStringBuilder.append(parseTime(String.valueOf(finishedTimeEnd)));
+    }
+
+    String timeFilterString = timeFilterStringBuilder.toString();
+
+
+    String sql;
+    StringBuilder sqlBuilder = new StringBuilder();
+    sqlBuilder.append("select count(id) as num_of_applications, count(distinct(job_exec_id)) as num_of_jobs, count(distinct(flow_exec_id)) as num_of_flows, sum(resource_used) as total_resource_used, sum(resource_wasted) as total_resource_wasted, sum(finish_time) - sum(start_time) as execution_time, sum(total_delay) as total_delay from yarn_app_result where");
```

Comment:
It is not possible to have aggregation functions using ebean. This page is currently for internal purposes and should not affect any other functionality.